lesson_1

alfacentaur85 opened the pull request:
lesson_1

alfacentaur85 committed:
lesson_1

gkurbesov commented on lesson1/ListWrapper/ListWrapper.cs:
```diff
@@ -0,0 +1,47 @@
+    
+    public class ListWrapper<T> : List<T>
+    {
+        private static readonly object _obj = new object();
```
Грубая ошибка, которая снизит производительность работы при большом кол-ве использовании экземпляров.
Статический объект синхронизации будет использоваться всем инстансами обертки, про приведёт к большой очереди ожидания. Необходимо убрать статический модификатор

gkurbesov commented on lesson1/ListWrapper/ListWrapper.cs:
```diff
@@ -0,0 +1,47 @@
+        {
+            lock (_obj)
+            {
+                List<T> lst = new List<T>();
+                
+                lst = this;
+
+                lst.Add(item);
```
можно заменить одним `this.Add(item);`

gkurbesov commented on lesson1/ListWrapper/ListWrapper.cs:
```diff
@@ -0,0 +1,47 @@
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
```
Нарушение принципа ответственности. Обертка для работы с коллекцией должна работать только с коллекцией

gkurbesov commented on lesson1/ListWrapper/ListWrapper.cs:
```diff
@@ -0,0 +1,47 @@
+
+                    lst = this;
+
+                    lst.Remove(item);
```
`return this.Remove(item);`

gkurbesov commented on lesson1/WPF/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,102 @@
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        static int sleepTime;
```
Убрать статик. поле, которое может разделятся несколькими формами и менять значение приведет к ошибке.
у вас получается, что имеется переменная, которая может поменяться из нескольких одинаковых форм, а изменение в одной форме повлечет изменения работы во всех формах

gkurbesov commented on lesson1/WPF/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,102 @@
+
+            sleepTime = Convert.ToInt32(slider.Value);
+
+            textBoxSlider.Text = (Convert.ToInt32(slider.Value)).ToString();
```
Вы уже выполнили конвертацию выше, нет смысла делать это несколько раз

gkurbesov commented on lesson1/WPF/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,102 @@
+        {
+            if (textBoxSlider != null)
+            {
+                textBoxSlider.Text = (Convert.ToInt32(((Slider)sender).Value)).ToString();
```
На форму выводите, а переменную `sleepTime `не меняете

alfacentaur85 committed:
Lesson2_mod2

Никак не получилось у меня не создавать в пуле новый thread, т.к. нигде не нашел информации, как оставить поток живым чтобы он в завершенные не переходил и назначить ему новый Action на выполнение. Совет в инете был Thread на Task заменить. Я попробовал, но получилось ещё хуже, чем со thread.  Вот эти два вопроса оказались для меня неразрешимыми...
